src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -355,6 +356,17 @@ public Task<DocumentDatabase> TryGetOrCreateResourceStore(StringSegment database
 
                 if (DatabasesCache.TryGetValue(databaseName, out var database))
                 {
+                    if (database.IsFaulted && database.Exception.InnerExceptions.Count > 0)
+                    {
+                        foreach (var inner in database.Exception.InnerExceptions)
+                        {
+                            // If a database was unloaded, this is what we get from DatabasesCache. 
+                            // We want to keep the exception there until UnloadAndLockDatabase is disposed.
+                            if (inner.GetType() == typeof(DatabaseDisabledException))
```

Comment:
`inner is DatabaseDisabledException`, but this looks _incredibly_ fragile.
We can use `ExtractSingleInnerException` here.